src/middlewares/permissions.js

```diff
@@ -106,36 +106,41 @@ class PermissionMiddlewareCreator {
   // whithin the registered scope
   _ensureRecordIdsInScope() {
     return async (request, response, next) => {
-      const model = this.modelsManager.getModelByName(this.collectionName);
-      const attributes = PermissionMiddlewareCreator._getRequestAttributes(request);
-      const { timezone } = request.query;
-
-      // if performing a `selectAll` let the `getIdsFromRequest` handle the scopes
-      if (attributes.allRecords) return next();
-
-      const { idField } = Schemas.schemas[this.collectionName];
-
-      // TODO: scope smartAction calls properly on table with composite primary keys
-      if (idField === 'forestCompositePrimary') return next();
-
-      const targetRecordIds = attributes.ids;
-      const checkIdsFilter = JSON.stringify({
-        field: idField,
-        operator: 'in',
-        value: targetRecordIds,
-      });
+      try {
+        // if performing a `selectAll` let the `getIdsFromRequest` handle the scopes
+        const attributes = PermissionMiddlewareCreator._getRequestAttributes(request);
+        if (attributes.allRecords) {
+          return next();
+        }
+
+        // Otherwise, check that all records are within scope.
+        const { primaryKeys } = Schemas.schemas[this.collectionName];
+        const filters = JSON.stringify(primaryKeys.length === 1
+          ? { field: primaryKeys[0], operator: 'in', value: attributes.ids }
+          : {
+            aggregator: 'or',
```

Comment:
I don't understand why those lines show as untested in the code coverage
Do you get it?

Reply:
I don't, my guess would be thatCC does not compute inline well ternaries like this one 🤷